src/catalog/database_catalog.cpp

```diff
@@ -2557,6 +2589,200 @@ proc_oid_t DatabaseCatalog::GetProcOid(common::ManagedPointer<transaction::Trans
   return ret;
 }
 
+template <typename Column, typename ClassOid, typename ColOid>
+void DatabaseCatalog::CreateColumnStatistic(const common::ManagedPointer<transaction::TransactionContext> txn,
```

Comment:
Why dont we have tests of all these functions. This is supposed to be the heart of the PR, while we never see these running in action. You created a separate table in the test and added things to it manually. You should use these functions, so in the future someone adding functionality in the optimizer can see the test and know how to do things.